gax-java/gax/src/main/java/com/google/api/gax/rpc/ApiClientHeaderProvider.java

```diff
@@ -41,13 +41,18 @@
 public class ApiClientHeaderProvider implements HeaderProvider, Serializable {
   private static final long serialVersionUID = -8876627296793342119L;
   static final String QUOTA_PROJECT_ID_HEADER_KEY = "x-goog-user-project";
+  static final String PROTOBUF_HEADER_VERSION_KEY = "protobuf";
 
   public static final String API_VERSION_HEADER_KEY = "x-goog-api-version";
+  private static final String protobufVersionAppendValue =
+      "--" + PROTOBUF_HEADER_VERSION_KEY + "-" + GaxProperties.getProtobufVersion();
+  private static String tokensToAppendProfobufVersionTo = "";
 
   private final Map<String, String> headers;
 
   protected ApiClientHeaderProvider(Builder builder) {
     ImmutableMap.Builder<String, String> headersBuilder = ImmutableMap.builder();
+    tokensToAppendProfobufVersionTo = "(gccl|gapic).*";
```

Comment:
`tokensToAppendProtobufVersionTo` is a `static` field being modified multiple times in an instance constructor.

Although one solution is to remove the static keyword, please see if we can refactor this to not need to use a class field to track temporary constructor state. For example, a final `apiClientHeaderValue.replaceFirst(regex, replacement)` after the string has been constructed ([see example here](https://www.w3schools.com/java/ref_string_replacefirst.asp)). (If efficiency is a concern, we should consider a solution that doesn't require performing a regex evaluation on each appendToken call.)

Reply:
good call,  i think solution is much cleaner now